vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -99,6 +101,8 @@ def initiate_api_connection(self):
             self.api = VSphereAPI(self._config, self.log)
             self.log.debug("Connected")
         except APIConnectionError:
+            # Clear the API connection object if the authentication fails
+            self.api = cast(VSphereAPI, None)
```

Comment:
Why not just set `self.api = None`?

Reply:
`self.api` has a type of `VSphereAPI`, so the code has to set it to something that is of type of `VSphereAPI`